packages/api/amplify_api/lib/src/graphql/web_socket/blocs/web_socket_bloc.dart

```diff
@@ -343,6 +350,25 @@ class WebSocketBloc with AWSDebuggable, AmplifyLoggerMixin {
     ]);
   }
 
+  void _emitHubEvent(WebSocketState state) {
+    logger.debug('Emitting hub event: $state');
+
+    if (state is ConnectingState) {
+      _hubEventsController.add(SubscriptionHubEvent.connecting());
+    } else if (state is ConnectedState) {
+      _hubEventsController.add(SubscriptionHubEvent.connected());
+    } else if (state is PendingDisconnect) {
+      _hubEventsController.add(SubscriptionHubEvent.pendingDisconnect());
+    } else if (state is DisconnectedState) {
+      // Once disconnected we throw away the bloc and need to close hub controller
+      _hubEventsController
+        ..add(SubscriptionHubEvent.disconnected())
+        ..close();
```

Comment:
Let's keep this method single-purpose and only close the controller when the rest of them are closed